src/tests/ftest/util/server_utils.py

```diff
@@ -49,6 +53,189 @@ class ServerFailed(Exception):
+        kill_server(self.hosts)
+
+        # Clean up any files that exist on the hosts
+        clean_server(self.hosts)
+
+        # Ensure the environment for the daos server on each host
+        run_agent(path, self.hosts)
+
+        # Create the hostfile
+        self.hostfile = write_host_file(self.hosts, path, slots)
+
+    def update_configuration(self, basepath):
+        """Update the config parameter with a yaml file.
+
+        Args:
+            basepath (str): DAOS install basepath
+        """
+        self.config.update(create_server_yaml(basepath), "server.config")
+
+    def get_param_names(self):
+        """Get a sorted list of daos_server command parameter names."""
+        names = self.get_attribute_names(FormattedParameter)
+        names.extend(["request", "action"])
+        return names
+
+    def get_params(self, test, path="/run/daos_server/*"):
+        """Get values for all of the server command params using a yaml file.
+
+        Sets each BasicParameter object's value to the yaml key that matches
+        the assigned name of the BasicParameter object in this class. For
+        example, the self.block_size.value will be set to the value in the yaml
+        file with the key 'block_size'.
+
+        Args:
+            test (Test): avocado Test object
+            path (str, optional): yaml namespace.
+                Defaults to "/run/daos_server/*".
+
+        """
+        super(ServerCommand, self).get_params(test, path)
+
+    def get_launch_command(self, manager, uri=None, env=None):
+        """Get the process launch command used to run daos_server.
+
+        Args:
+            manager (str): mpi job manager command
+            uri (str, optional): path to uri file. Defaults to None.
+            env (dict, optional): dictionary on env variable names and values.
+                Defaults to None.
+
+        Raises:
+            ServerFailed: if the specified job manager is unsupported.
+
+        Returns:
+            str: daos_sever launch command
+
+        """
+        if manager is None:
+            # Run locally
+            return self.__str__()
+
+        elif manager.endswith("orterun"):
+            args = []
+            if env:
+                assign_env = [
+                    "{}={}".format(key, val) for key, val in env.items()]
+                args = [
+                    "-np {}".format(len(self.hosts)),
+                    "-hostfile {}".format(self.hostfile),
+                    "--enable-recovery",
+                ]
+                args.extend(["-x {}".format(assign) for assign in assign_env])
+            if uri is not None:
+                args.append("--report-uri {}".format(uri))
+
+            return "{} {} {}".format(manager, " ".join(args), self.__str__())
+        else:
+            raise ServerFailed("Unsupported job manager: {}".format(manager))
+
+    def start(self, manager, verbose=True, env=None, sudo=None, timeout=600):
```

Comment:
Isn't this actually `run()` like: https://github.com/daos-stack/daos/blob/dd5111672e3983f52f42176ded65f0fee2c2d3be/src/tests/ftest/util/dmg_utils.py#L84 albeit with the added complication of a `manager`.

Reply:
start() runs a cmd in the background. The run() methods starts the cmd and waits for it to finish.

Reply:
Interesting contrast.  I might have chosen different names, but that's subjective.